openedx/features/content_type_gating/tests/test_access.py

```diff
@@ -0,0 +1,108 @@
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+
+
+@ddt.ddt
+class TestProblemTypeAccess(SharedModuleStoreTestCase):
+
+    @classmethod
+    def setUpClass(self):
+        super(TestProblemTypeAccess, self).setUpClass()
+        self.factory = RequestFactory()
+        self.course = CourseFactory.create(run='testcourse1', display_name='Test Course Title')
+        CourseModeFactory.create(course_id=self.course.id, mode_slug='audit')
+        CourseModeFactory.create(course_id=self.course.id, mode_slug='verified')
+        with self.store.bulk_operations(self.course.id):
+            self.chapter = ItemFactory.create(
+                parent=self.course,
+                display_name='Overview'
+            )
+            self.welcome = ItemFactory.create(
+                parent=self.chapter,
+                display_name='Welcome'
+            )
+            ItemFactory.create(
+                parent=self.course,
+                category='chapter',
+                display_name='Week 1'
+            )
+            self.chapter_subsection = ItemFactory.create(
+                parent=self.chapter,
+                category='sequential',
+                display_name='Lesson 1'
+            )
+            chapter_vertical = ItemFactory.create(
+                parent=self.chapter_subsection,
+                category='vertical',
+                display_name='Lesson 1 Vertical - Unit 1'
+            )
+            self.problem = ItemFactory.create(
+                parent=chapter_vertical,
+                category='problem',
+                display_name='Problem - Unit 1 Problem 1',
+                graded=True,
+            )
+            self.dragndrop = ItemFactory.create(
+                parent=chapter_vertical,
+                category='drag-and-drop-v2',
+                display_name='Drag Problem - Unit 1 Problem 2',
+                graded=True,
+            )
+            self.ora = ItemFactory.create(
+                parent=chapter_vertical,
+                category='openassessment',
+                display_name='ORA - Unit 1 Problem 3',
+                graded=True,
+            )
+            self.done = ItemFactory.create(
+                parent=chapter_vertical,
+                category='done',
+                display_name='Done - Unit 1 Problem 4',
+                graded=True,
+            )
+            self.edx_sga = ItemFactory.create(
+                parent=chapter_vertical,
+                category='edx_sga',
+                display_name='SGA - Unit 1 Problem 5',
+                graded=True,
+            )
+
+    def setUp(self):
+        super(TestProblemTypeAccess, self).setUp()
+        self.audit_user = UserFactory.create()
+        self.enrollment = CourseEnrollmentFactory.create(user=self.audit_user, course_id=self.course.id, mode='audit')
+
```

Comment:
Is there a reason a new user is created and enrolled before every test? could this be in the setUpClass method?

Reply:
@cpennington directed me to do this as our user factory is based in a Django db that is different from the way we instantiate and store our course objects so the class setup method would not work the same way for user creation and enrollment? I didn't understand this fully so maybe Cale can weigh in again here.

Reply:
Yeah, I don't think that django `TestCase` is prepared to handle `setUpClass` usage. (Or at least, I couldn't find anything to that effect in their docs. I haven't spelunked the code to verify for sure, though).